src/interchange/src/avro/schema.rs

```diff
@@ -7,13 +7,42 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
+//! Conversion from Avro schemas to Materialize `RelationDesc`s.
+//!
+//! A few notes for posterity on how this conversion happens are in order.
+//!
+//! If the schema is an Avro record, we flatten it to its fields, which become the columns
+//! of the relation.
+//!
+//! Each individual field is then converted to its SQL equivalent. For most types, this
+//! conversion is the obvious one. The only non-trivial counterexample is Avro unions.
```

Comment:
what about nested types? or does avro not support that?

Reply:
The conversion proceeds recursively!